### PR TITLE
Add Archive Page + pagination pattern on all template

### PR DESCRIPTION
**Pull request by gabrielcaires**
Issue #10 

##  Introduced changes
* Create a pagination pattern to support i18n
* Apply the pattern on Search/Archive and Index templates.

## Testing instructions:
* Go to the archive/search/index page
* Check if the pagination matches the design

Screen Shot 2022-10-14 at 16 51 49

### CODE REVIEW

**Comment by gabrielcaires** [comment]
@Luchadores I didn't find a suggestion to display the current page, so I added an underline as you can see here:

Screen Shot 2022-10-14 at 16 51 49

Is it fine? do you have any other suggestions?

**Comment by Luchadores** [comment]
That's really nice, thanks, Gabriel!

File: style.css [hunk 1]
```diff
@@ -227,3 +227,17 @@ body .is-layout-flow .wp-block-group {
...
+/*
+ * Pagination styles
+ */
+.wp-block-query-pagination .wp-block-query-pagination-numbers > *:not(:last-child) {
+	margin-right: clamp(1.44rem, 1.91vw + 0.97rem, 2.5rem);
+}
+
+.wp-block-query-pagination .wp-block-query-pagination-numbers > *:not(:last-child) {
+	margin-right: clamp(1.44rem, 1.91vw + 0.97rem, 2.5rem);
+}
```
Comment (Imran92) [inline style.css]: A slip of keyboard probably, this style is added twice

File: templates/archive.html [hunk 2]
```diff
@@ -1,33 +1,24 @@
-<!-- wp:template-part {"slug":"header","tagName":"header"} /-->
+<!-- wp:template-part {"slug":"header","theme":"sensei-theme","tagName":"header"} /-->
 
-<!-- wp:query {"tagName":"main","layout":{"inherit":true}} -->
+<!-- wp:query {"queryId":1,"query":{"perPage":10,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":true,"taxQuery":null,"parents":[]},"tagName":"main","displayLayout":{"type":"flex","columns":3},"layout":{"inherit":true}} -->
```
Comment (Imran92) [inline templates/archive.html]: As this is a template, should we remove the queryId?

Comment (gabrielcaires) [reply templates/archive.html]: Fixed here 43cd4538f55fc7dc4502f9a519e1bf2fb97a99d9

File: style.css (hunk 1 above)
Comment (gabrielcaires) [reply style.css]: Fixed here [0243544](https://github.com/Automattic/sensei-theme/pull/39/commits/0243544893ca778bf5bdd8fde042541721a8da7d)

File: patterns/pagination.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+/**
+ * Title: Default Pagination 
+ * Slug: course/pagination
+ * Inserter: yes
```
Comment (donnapep) [inline patterns/pagination.php]: I don't think we should make this available in the inserter, as it only makes sense to use pagination inside the Query block.

File: style.css [hunk 4]
```diff
@@ -227,3 +227,14 @@ body .is-layout-flow .wp-block-group {
...
 }
+
+/*
+ * Pagination styles
+ */
+.wp-block-query-pagination .wp-block-query-pagination-numbers > *:not(:last-child) {
+	margin-right: clamp(1.44rem, 1.91vw + 0.97rem, 2.5rem);
+}
+
+.wp-block-query-pagination .page-numbers.current{
```
Comment (donnapep) [inline style.css]:
```suggestion
.wp-block-query-pagination .page-numbers.current {
```

File: patterns/pagination.php [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ */
+?>
+
+<!-- wp:group {"layout":{"inherit":false,"type":"constrained"}} -->
+<div class="wp-block-group">
+  <!-- wp:query-pagination {"layout":{"type":"flex","justifyContent":"space-between"}} -->
+
+  <!-- wp:query-pagination-numbers /-->
+
+  <!-- wp:query-pagination-next {"label":"NEXT"} /-->
```
Comment (donnapep) [inline patterns/pagination.php]: If we're transforming this to uppercase later in `theme.json`, does this need to be capitalized here? I'm just thinking of translators who may think it's weird when they see this string is in all caps.

File: theme.json [hunk 6]
```diff
@@ -401,6 +401,28 @@
...
+					"fontFamily": "var(--wp--preset--font-family--league-gothic)",
+					"fontSize": "var(--wp--preset--font-size--medium)"
+				}
+			},
+			"core/post-featured-image": {
+				"border": {
+					"color": "var(--wp--preset--color--foreground)",
+					"radius": "0.25rem",
+					"style": "solid",
+					"width": "2px"
```
Comment (donnapep) [inline theme.json]: This should be `1px` as per the design.

File: theme.json [hunk 7]
```diff
@@ -401,6 +401,28 @@
...
+					"link": {
+						"typography": {
+							"textDecoration": "none",
+							"textTransform": "uppercase"
+						}
+					}
+				},
+				"typography": {
+					"fontFamily": "var(--wp--preset--font-family--league-gothic)",
+					"fontSize": "var(--wp--preset--font-size--medium)"
```
Comment (donnapep) [inline theme.json]:
```suggestion
					"fontSize": "var(--wp--preset--font-size--medium)",
					"letterSpacing": "0.01em"
```

**Comment by donnapep** [comment]
The pagination doesn't seem to be working correctly in all cases. Here's how to reproduce:

- Create 2 posts and assign the same category (e.g. "Test") to both.
- In Settings > Reading, set "Blog pages show at most" to 2.
- View the category archive page (e.g. http://senseitheme.local/category/test/)
- I see 2 page numbers, even though all posts are visible, as well as the "Next" link. I wouldn't expect to see any page numbers or links here since there are no more posts to show.

Screen Shot 2022-10-28 at 8 36 46 AM

**Review by donnapep — COMMENTED** [review]
Found some smallish things in code.

File: patterns/pagination.php (hunk 3 above)
Comment (gabrielcaires) [reply patterns/pagination.php]: Fixed here 768c45c283a7102034ce7172d08e9aa442e0eb2e

File: theme.json (hunk 7 above)
Comment (gabrielcaires) [reply theme.json:618]: Fixed here 2b058547c7725ff915984e661af8d6d2d9b3d29b

File: theme.json (hunk 6 above)
Comment (gabrielcaires) [reply theme.json]: fixed here 1d3ddd0de288fd4c58c0532d1ae2d8ee6af3a96f

File: theme.json (hunk 7 above)
Comment (gabrielcaires) [reply theme.json]: Fixed here ba4fba0700d06c6e10c23ce3b738164cbfa549a2

File: patterns/pagination.php (hunk 5 above)
Comment (gabrielcaires) [reply patterns/pagination.php]: Fixed here 0a275ef8803fc965c911b3a679e83d66736fe804

**Comment by gabrielcaires** [comment]
> The pagination doesn't seem to be working correctly in all cases. Here's how to reproduce:
> 
> * Create 2 posts and assign the same category (e.g. "Test") to both.
> * In Settings > Reading, set "Blog pages show at most" to 2.
> * View the category archive page (e.g. http://senseitheme.local/category/test/)
> * I see 2 page numbers, even though all posts are visible, as well as the "Next" link. I wouldn't expect to see any page numbers or links here since there are no more posts to show.
> 
> Screen Shot 2022-10-28 at 8 36 46 AM

@donnapep Could you please test this scenario again? I was not able to reproduce it.

**Comment by Imran92** [comment]
Hi @gabrielcaires, I tried to reproduce the issue @donnapep mentioned above and I could reproduce it. The pagination did  not work as expected for me too, if you have 3 posts, and have set to see 2 posts per page, clicking on the 2nd page in the pagination won't take you to the second page, rather it was refreshing the first page again.

I tried to look into a bit and it turned out if you split the pagination in a separate pattern and reference that pattern inside query loop (as we did here in the archive page template), it doesn't work. It shows the pagination block as expected, but the functionality breaks. So instead of having a separate pattern file for pagination, if we move the markup of the pagination pattern to the archive template, it will hopefully work. If it works for you too, then I think we can also remove the pattern file for the pagination for now.

Hope this helps! <3

**Comment by gabrielcaires** [comment]
> Hi @gabrielcaires, I tried to reproduce the issue @donnapep mentioned above and I could reproduce it. The pagination did not work as expected for me too, if you have 3 posts, and have set to see 2 posts per page, clicking on the 2nd page in the pagination won't take you to the second page, rather it was refreshing the first page again.
> ...

Fixed here as you suggested @Imran92 
[a1762d5](https://github.com/Automattic/sensei-theme/pull/39/commits/a1762d5466c25025566997b698c45f1f2751227c)

**Comment by donnapep** [comment]
Works well on the archive page, but the pagination pattern is still being used in `index.html` and `search.html`.

**Comment by gabrielcaires** [comment]
>  Works well on the archive page, but the pagination pattern is still being used in index.html and search.html.

@donnapep  Fixed here b2df18944090400d3935e7b206b773facb420d12

**Comment by Imran92** [comment]
The dimension of the featured image inside the query loop here doesn't match the design in the archive template

image

**Comment by Imran92** [comment]
The design for mobile looks broken. I've attached a screenshot below. Show we don't have a design specifically for archive page in mobile devices, we have design for similar pages that has query loop inside that have design for mobile. I'm attaching an screenshot of that as well.

##**Archive page in mobile**

image

##**Similar page design with query loop and pagination for mobile - from figma**

image

**Comment by Imran92** [comment]
I've found a few more issues and fixed them [here](https://github.com/Automattic/course/pull/39/commits/e52a22cdee3923f668b47bed7a7e3b1fabc6dd35)

Some of the fixed items are -
- Title font size
- Spacing between the course content items
- Fix pagination taking full width in mobile
- Spacing between header-footer and content
- Featured image size
- Fixed the mobile view
- Fixed line height for all post content blocks (date, title, excerpt etc)

Now they look as follows-

## Desktop


Screen Shot 2022-11-10 at 3 09 25 AM


## Mobile

Screen Shot 2022-11-10 at 3 22 37 AM

**Review by Imran92 — APPROVED** [review]
I've found some issues and have fixed them and made some updates too. Now LGTM

**Pull request merged**